### PR TITLE
PEN-676: [NVDA | Firefox] Article Landing Page > Author's Profile: Social media profile icons are read inappropriately via NVDA 2019 on Firefox.

### PR DESCRIPTION
**Pull request by vinnie-olsen**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Add 'aria-label' to author-bio social links

## Jira Ticket
- [PEN-676](https://arcpublishing.atlassian.net/browse/PEN-676)

## Acceptance Criteria
_copy from ticket_

## Test Steps

1. In `Fusion-News-Theme` repo, checkout `master` & `git pull`
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/author-bio-block` in `Fusion-News-Theme`
4. We are planning on turning these accessibility tickets over to the auditing firm that reported the issues in the first place. We don't have access to these tools and most are either paid or Windows-only. For that reason, just take a look over the code to make sure it makes sense and check http://localhost/local/2020/01/14/tampa-reinvents-itself-to-show-off-its-shores/?_website=the-gazette to make sure that the `aria-label` attributes are now present on the social links in the author bio (see screenshot below). 

image

### Before
No `aria-label` attributes were present on the social links in the author bio

### After
There are now `aria-label` attributes with descriptive content present on the social links in the author bio

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [X] Confirmed all the test steps a reviewer will follow above are working. 
- [X] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [X] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [X] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [X] Confirmed this PR has unit test files
  - [X] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [X] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr. 
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/author-bio-block/features/author-bio/default.jsx [hunk 1]
```diff
@@ -98,167 +98,189 @@ const AuthorBio = () => {
...
             let socialButton;
+            let linkTitle;
             const constructedURL = constructSocialURL(socialLink.site, socialLink.url);
 
+            const MediaLink = ({ children, label, ...otherProps }) => (
+              <MediaLinksStyled
+                href={constructedURL}
+                target="_blank"
+                rel="noreferrer noopener"
+                aria-label={`Click ${label} link`}
```
Comment (matthewroach) [inline blocks/author-bio-block/features/author-bio/default.jsx]: You shouldn't use the word "link" within the description as this is handled by a screen reader and causes duplication, the same goes for "click" uses are told they are on a link and this also implies you need to "click" it, which isn't always the case for non mouse users


This is what a screen reader will read out when they see this link now - "link, image, Click Twitter link"
PEN-676 VoiceOver link value

Don't worry too much about the image part, that is going to be taken care of in the SVG update ticket.

The idea here is we want to provide meaningful text to someone who is unable to gain the same visual experience we have from looking at the page.
Something like - [Vinnie Olsen] on [Twitter] - and replacing the name with the author and twitter with the correct value.

**Review by matthewroach — CHANGES_REQUESTED** [review]

File: blocks/author-bio-block/features/author-bio/default.jsx (hunk 1 above)
Comment (vinnie-olsen) [reply blocks/author-bio-block/features/author-bio/default.jsx]: @matthewroach updated the label, refactored a little and added more unit test coverage 👍  Let me know if that looks better!

**Review by matthewroach — APPROVED** [review]
Looks good to me, nice work @vinnie-olsen

**Comment by vinnie-olsen** [comment]
Awesome thanks @matthewroach !

**Pull request merged**